lib/core/include/irods/irods_load_plugin.hpp

```diff
@@ -94,6 +101,274 @@ namespace irods
 
     } // resolve_plugin_path
 
+    /**
+     * \brief attempt to parse DT_FLAGS_1 to identify flags to pass to dlopen()
+     *
+     * \since 4.3.1
+     **/
+    // NOLINTNEXTLINE(readability-function-cognitive-complexity) this is technically a parser
+    inline int get_plugin_rtld_flags(const std::string& _plugin_path) noexcept
```

Comment:
Consider adding comments to the most important parts of this function. It's just nice not having to find documentation for all the scary data structures to understand what is happening here.

Reply:
Yes. Please mark as resolved.